src/main/scala/com/github/anicolaspp/spark/sql/writing/MapRDBCleaner.scala

```diff
@@ -0,0 +1,15 @@
+package com.github.anicolaspp.spark.sql.writing
+
+import org.ojai.store.{DocumentStore, DriverManager}
+
+object MapRDBCleaner {
+
+  def clean(ids: Set[String], table: String): Unit = {
+
+    val connection = DriverManager.getConnection("ojai:mapr:")
+
+    val store: DocumentStore = connection.getStore(table)
+
+    ids.foreach(store.delete)
```

Comment:
In a normal rollback this should be fine since the only the confirmed written ids are in the `ids` value. But if one of the records to be deleted has been removed by an external application then the `store.delete` should throw an exception. It might need to be caught and ignored here.
One way to prevent a try catch here is to use the `checkAndDelete` method although that might actually create more overhead.